singleton/controller.go

```diff
@@ -36,3 +36,24 @@ func Source() source.Source {
 		},
 	})
 }
+
+// In response to Requeue: True being deprecated via: https://github.com/kubernetes-sigs/controller-runtime/pull/3107/files
+// This uses a bucket and per item delay but the item will be the same because the key is the controller name..
+// This implements the same behavior as Requeue: True.
+type SingletonRateLimiter struct {
```

Comment:
Have we thought about throwing this in `reasonable`? Just thinking about it, I figure that this would be used more frequently than just in the singleton controller -- seems like this is an item rate limiter that would determine what the RequeueAfter value should be -- how does this work generally across all types of controllers?